cmd/csvlint/main.go

```diff
@@ -22,16 +24,17 @@ func main() {
 		printHelpAndExit(0)
 	}
 
-	var comma rune
-	switch *delimiter {
-	case "comma":
-		comma = ','
-	case "tab":
-		comma = '\t'
-	default:
-		fmt.Printf("unrecognized delimiter '%s'\n\n", *delimiter)
+	if flag.NFlag() > 0 {
+		fmt.Fprintln(os.Stderr, "Warning: not using defaults, may not validate CSV to RFC 4180")
+	}
+
+	converted_delimiter, err := strconv.Unquote(`'` + *delimiter + `'`)
```

Comment:
as for why we need it, it's a little awkward to be able to read in things like `\t':
https://groups.google.com/forum/#!topic/golang-nuts/jYEvC9C3tGo

Without a little bit of hoop-jumping, the input will have to be unicode:
`csvlint --delimiter="    "`
Where those spaces would be an actual tab (it's _really_ annoying to actually paste a tab!!)

Unquoting takes care of this for us. Accepting all code golf, though, it's a bit awkward I think... :-/